server/container_create_linux.go

```diff
@@ -1249,15 +1255,50 @@ func (s *Server) mountImage(ctx context.Context, imageRef, mountLabel string) (h
 	log.Debugf(ctx, "Image ID to mount: %v", id)
 
 	options := []string{"ro", "noexec", "nosuid", "nodev"}
-	mountPoint, err := s.Store().MountImage(id, options, mountLabel)
+	mountPoint, err := s.Store().MountImage(id, options, "")
 	if err != nil {
 		return "", "", fmt.Errorf("mount storage: %w", err)
 	}
-
 	log.Infof(ctx, "Image mounted to: %s", mountPoint)
+
+	const overlay = "overlay"
+	specgen.AddMount(rspec.Mount{
+		Type:        overlay,
+		Source:      overlay,
+		Destination: containerPath,
+		Options: []string{
+			"lowerdir=" + mountPoint + ":" + imageVolumesPath,
```

Comment:
We may need to get the kind of the storage driver that is used by the cri-o. I'm not sure whether it can work properly on other backends, i.e. zfs, btrfs, etc.

Reply:
I don't think so, the `lowerdir=` specification will mainly work for overlay. But overlay should be available most of the times, right?